src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,763 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
             return dbResultSetRow;
         }
 
+        /// <summary>
+        /// Performs the given GraphQL create mutation operation.
+        /// </summary>
+        /// <param name="entityName">Name of the top level entity</param>
+        /// <param name="parameters">Mutation parameter arguments</param>
+        /// <param name="sqlMetadataProvider">SqlMetadaprovider</param>
+        /// <param name="context">Hotchocolate's context for the graphQL request.</param>
+        /// <param name="multipleInputType">Boolean indicating whether the create operation is for multiple items.</param>
+        /// <returns>Primary keys of the created records (in the top level entity).</returns>
+        /// <exception cref="DataApiBuilderException"></exception>
+        private List<IDictionary<string, object?>> PerformMultipleCreateOperation(
+                string entityName,
+                IMiddlewareContext context,
+                IDictionary<string, object?> parameters,
+                ISqlMetadataProvider sqlMetadataProvider,
+                bool multipleInputType = false)
+        {
+            string fieldName = multipleInputType ? MULTIPLE_INPUT_ARGUEMENT_NAME : SINGLE_INPUT_ARGUEMENT_NAME;
+
+            // Parse the hotchocolate input parameters into .net object types
+            object? parsedInputParams = GQLMultipleCreateArgumentToDictParams(context, fieldName, parameters);
+
+            if (parsedInputParams is null)
+            {
+                throw new DataApiBuilderException(message: "The input for multiple create mutation operation cannot be null",
+                                                  statusCode: HttpStatusCode.BadRequest,
+                                                  subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+            }
+
+            // List of Primary keys of the created records in the top level entity.
+            // Each dictionary in the list corresponds to the PKs of a single record.
+            // For point multiple create operation, only one entry will be present.
+            List<IDictionary<string, object?>> primaryKeysOfCreatedItemsInTopLevelEntity = new();
+
+            if (!parameters.TryGetValue(fieldName, out object? param) || param is null)
+            {
+                throw new DataApiBuilderException(message: $"Mutation Request should contain the expected argument: {fieldName} in the input",
+                                                  statusCode: HttpStatusCode.BadRequest,
+                                                  subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+            }
+
+            if (multipleInputType)
+            {
+                int idx = 0;
```

Comment:
what is this `idx` tracking. place this nearer to where it is actually used.

Reply:
`idx` needs to initialized with `0` before the following `foreach` loop. Moved it just before the loop and added some comments on where `idx` is used.

Reply:
Renamed to `itemsIndex`